apps/whois/plugin.pm

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright 2021 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors : Pedro Manuel Santos Delgado - lunik
+#
+package apps::whois::plugin;
+
+use strict;
+use warnings;
+use base qw(centreon::plugins::script_simple);
+
+sub new {
+    my ($class, %options) = @_;
+
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+
+    $self->{version} = '1.0';
+    $self->{modes} = {
+        'whois'               => 'apps::whois::mode::whois'
+    };
+
```

Comment:
Use the custom cli mode which is the last standard. 

See: https://github.com/centreon/centreon-plugins/blob/5216525e8b440a1b40c4ab5ab5e811e6c3dd8974/apps/nmap/cli/plugin.pm#L37

Reply:
@PedroMSantosD look at this comment this is why the option{custom}->execute_command fails